lib/pbench/cli/server/shell.py

```diff
@@ -73,23 +73,26 @@ def keycloak_connection(config: PbenchServerConfig, logger: Logger):
         sys.exit(1)
 
     session = requests.Session()
-    # The connection check will retry thrice unless successful, viz.,
-    # [0.0s, 20.0s, 40.0s]. urllib3 will sleep for:
+    # The connection check will retry multiple times unless successful, viz.,
+    # [0.0s, 4.0s, 8.0s, 16.0s, ..., 256.0s]. urllib3 will sleep for:
```

Comment:
Not to be picky or anything, but according to the [reference](https://urllib3.readthedocs.io/en/latest/reference/urllib3.util.html#module-urllib3.util.retry) at line 81, the maximum backoff time (by default) is `120`, not `256`.  😉 

So, the 8 retries will delay 0, 4, 8, 16, 32, 64, 120, and 120 seconds, respectively, or a total of 364s or just over 6 minutes...which is probably fine.

However, I might be inclined to specify a lower `backoff_max`, both to reduce the total time and to prevent us from waiting so long between tries at the end of the sequence.  (I'm thinking that a maximum of 60s would be about right -- that would cut the timeout to exactly 4 minutes.)

Reply:
You are right I should update the comment here.